geode-connectors/src/main/java/org/apache/geode/connectors/jdbc/internal/SqlHandler.java

```diff
@@ -54,6 +60,13 @@ public SqlHandler(InternalCache cache, String regionName,
     this.tableMetaDataManager = tableMetaDataManager;
     this.regionMapping = getMappingForRegion(configService, regionName);
     this.dataSource = getDataSource(dataSourceFactory, this.regionMapping.getDataSourceName());
+    try {
+      compareTableMetadataWithMapping(tableMetaDataManager, regionMapping, regionName);
+    } catch (Exception ex) {
+      System.out.println("Throwing Exception...");
+      throw new IllegalStateException(
```

Comment:
I think this should be a JdbcConnectorException instead of IllegalStateException. This exception can happen due to user error (they changed their table def after creating the mapping) so JdbcConnectorException fits and is more specific than IllegalStateException.

Reply:
I agree, that exception would fit better.